src/common/arrstr.cpp

```diff
@@ -759,47 +756,52 @@ struct wxStringFragment
 
 wxStringFragment GetFragment(wxString& text)
 {
-    static const wxRegEx reSpaceOrPunct(wxS("^([[:space:]]|[[:punct:]])+"));
-    // Limit the length to make sure the value will fit into a wxUint64
-    static const wxRegEx reDigit(wxS("^[[:digit:]]{1,19}"));
-    static const wxRegEx reLetterOrSymbol("^[^[:space:]|[:punct:]|[:digit:]]+");
-
     if ( text.empty() )
         return wxStringFragment();
 
-    wxStringFragment fragment;
-    size_t           length = 0;
+    // the maximum length of a sequence of digits that
+    // can fit into wxUint64 when converted to a number
+    static const size_t maxDigitSequenceLength = 19;
 
-    // In attempt to minimize the number of wxRegEx.Matches() calls,
-    // try to do them from the most expected to the least expected
-    // string fragment type.
-    if ( reLetterOrSymbol.Matches(text) )
-    {
-        if ( reLetterOrSymbol.GetMatch(NULL, &length) )
-        {
-            fragment.type = wxStringFragment::LetterOrSymbol;
-            fragment.text = text.Left(length);
-        }
-    }
-    else if ( reDigit.Matches(text) )
+    wxStringFragment         fragment;
+    wxString::const_iterator it;
+
+    for ( it = text.begin(); it != text.end(); ++it )
     {
-        if ( reDigit.GetMatch(NULL, &length) )
+        const wxUniChar        ch = *it;
```

Comment:
```suggestion
        const wxUniChar&       ch = *it;
```
probably with little impact, but in the end it should avoid another copy of each string.